plugins/woocommerce/tests/php/includes/gateways/paypal/class-wc-gateway-paypal-request-test.php

```diff
@@ -199,4 +198,193 @@ public function return_valid_site_id( $value ) {
+	 *
+	 * @param string      $paypal_order_id            The PayPal order ID.
+	 * @param string|null $expected_exception         The expected exception class, or null if no exception is expected.
+	 * @param string|null $expected_exception_message The expected exception message, or null if no exception is expected.
+	 * @return void
+	 *
+	 * @dataProvider provide_test_get_paypal_order_details
+	 */
+	public function test_get_paypal_order_details( string $paypal_order_id, ?string $expected_exception, ?string $expected_exception_message ) {
+		$response_mock_ref = function () use ( $paypal_order_id ) {
+			if ( 'ERROR_ID' === $paypal_order_id ) {
+				return new WP_Error( 'error', 'Some error occurred.' );
+			}
+
+			if ( 'FAILED_ID' === $paypal_order_id ) {
+				return array(
+					'response' => array(
+						'code' => 500,
+					),
+					'body'     => wp_json_encode(
+						array(
+							'name'    => 'SOME_ERROR',
+							'details' => array(
+								array( 'issue' => 'SOME_ISSUE' ),
+							),
+						)
+					),
+				);
+			}
+
+			return array(
+				'response' => array(
+					'code' => 200,
+				),
+				'body'     => wp_json_encode( array() ),
+			);
+		};
+
+		add_filter( 'pre_http_request', $response_mock_ref, 10, 2 );
+
+		if ( $expected_exception ) {
+			$this->expectException( $expected_exception );
+			$this->expectExceptionMessage( $expected_exception_message );
+		}
+
+		$request = new WC_Gateway_Paypal_Request( new WC_Gateway_Paypal() );
+
+		$response_data = $request->get_paypal_order_details( $paypal_order_id );
+
+		// Clean up the filter.
+		remove_filter( 'pre_http_request', $response_mock_ref );
+
+		if ( ! $expected_exception ) {
+			$this->assertIsArray( $response_data );
+		}
+	}
+
+	/**
+	 * Data provider for the `test_get_paypal_order_details` method.
+	 *
+	 * @return array
+	 */
+	public function provide_test_get_paypal_order_details(): array {
+		return array(
+			'order details error response'   => array(
+				'PayPal order ID'            => 'ERROR_ID',
+				'expected exception'         => Exception::class,
+				'expected exception message' => 'PayPal order details request failed: Some error occurred.',
+			),
+			'order details failed response'  => array(
+				'PayPal order ID'            => 'FAILED_ID',
+				'expected exception'         => Exception::class,
+				'expected exception message' => 'PayPal order details request failed. HTTP 500',
+			),
+			'order details success response' => array(
+				'PayPal order ID'            => 'SUCCESS_ID',
+				'expected exception'         => null,
+				'expected exception message' => null,
+			),
+		);
+	}
+
+	/**
+	 * Tests for the `get_paypal_order_purchase_unit_amount` method.
+	 *
+	 * @return void
+	 */
+	public function test_get_paypal_order_purchase_unit_amount(): void {
+		$order = WC_Helper_Order::create_order();
+		$order->set_cart_tax( 10 );
+		$order->set_shipping_tax( 0 );
+		$order->set_total( 60 );
+		$order->save();
```

Comment:
Not blocking: do we want additional test cases for further values, like non-zero discounts, zero tax amounts, etc. Maybe something worth adding in a follow-up.

Reply:
I went ahead and added those in https://github.com/woocommerce/woocommerce/pull/61359/commits/70d2f2d7dfbdd0e920c4c06b2ba2877052419e3e